crates/data/src/hyperliquid/model.rs

```diff
@@ -0,0 +1,49 @@
+use serde::Deserialize;
+use sqd_primitives::{BlockNumber, ItemIndex};
+
+
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Transaction {
+    pub transaction_index: ItemIndex,
+    pub user: String,
+    pub actions: Vec<serde_json::Value>,
+    pub raw_tx_hash: Option<String>,
+    pub error: Option<String>,
+}
+
+
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct BlockHeader {
+    pub height: BlockNumber,
+    pub hash: String,
+    pub proposer: String,
+    pub block_time: i64,
+}
+
+
+#[derive(Deserialize)]
+pub struct Block {
+    pub header: BlockHeader,
+    pub transactions: Vec<Transaction>,
+}
+
+
+impl sqd_primitives::Block for Block {
+    fn number(&self) -> BlockNumber {
+        self.header.height
+    }
+    
+    fn hash(&self) -> &str {
+        &self.header.hash
+    }
+
+    fn parent_number(&self) -> BlockNumber {
+        unimplemented!()
```

Comment:
Both `parent_number` and `parent_hash` are required. 

Without `parent_hash` not only hotblocks part, but also `query.parent_block_hash` checks will fail.

Reply:
what if `parent_hash` will return a struct which is equal to all `&str`? the struct could look like:
```rs
struct AlwaysEqual;

impl PartialEq<str> for AlwaysEqual {
    fn eq(&self, _other: &str) -> bool {
        true
    }
}
```
otherwise we need to sort of flags to all places where `parent_hash` is invoked